app/ui/browser/webview-controller.js

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import assert from 'assert';
+import { EventEmitter } from 'events';
+import { isUUID } from './browser-util';
+
+const getPageIndexById = (pages, id) => pages.findIndex(p => p.id === id);
+
+/**
+ * A utility class that extends EventEmitter to receive page IDs and
+ * emit events to be consumed by a WebView owner (in our case, page.jsx).
+ */
+class WebViewController extends EventEmitter {
+  constructor(getPages) {
+    super();
+    this.getPages = getPages;
+  }
+
+  navigateBack(pageId) {
+    const pages = this.getPages();
+    const index = getPageIndexById(pages, pageId);
+    assert(index >= 0, `Page ${pageId} not found in current state.`);
+    assert(isUUID(pageId), `Invalid page id: ${pageId}`);
+    assert(pages.get(index).canGoBack, 'Page cannot go back.');
+
+    this.emit('navigate-back', pageId);
```

Comment:
Consider also emitting `navigate-back-${pageId}`, to make it easier to listen to just one webview's message.  (And then wrapping `on` to hide that detail.)